File: BOJ/자료구조/10773.cpp
// 10773 제로
// https://www.acmicpc.net/problem/10773
#include <iostream>
#include <vector>

using namespace std;

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int K, val, answer = 0;
    vector<int> stack;

    cin >> K;
    while (K--)
    {
        cin >> val;
        if (val > 0) {
            stack.push_back(val);
        } else
        {
            if (stack.empty()) continue;
            stack.pop_back();
        }
    }
    
    while (!stack.empty())
    {
        answer += stack.back();
        stack.pop_back();
    }

    cout << answer;

    return 0;
}

File: BOJ/자료구조/10773.py
# 10773 제로
# https://www.acmicpc.net/problem/10773

def solution(k, inputList):
    stack = []

    for i in range(k):
        if inputList[i]:
            stack.append(inputList[i])

        elif not inputList[i] and len(stack) > 0:
            stack.pop()

    return sum(stack)


k = int(input())
inputList = [int(input()) for _ in range(k)]

print(solution(k, inputList))
